app/jobs/course/material/zip_download_job.rb

```diff
@@ -29,6 +29,6 @@ def publish_file(file, name)
     public_file = File.join(public_dir, name)
     FileUtils.cp(file, public_file)
 
-    File.join(downloads_dir, job.id, File.basename(public_file))
+    URI.encode(File.join('/', downloads_dir, job.id, File.basename(public_file)))
```

Comment:
This string should be encoded as url to handle the case that when there are white spaces inside file names.

Also redirect_to expects the url starting with `'/'`, my bad.